### PR TITLE
added possibility to move 3rd-party-modules into another directory

### PR DESCRIPTION
**Pull request by khassel**
Inspired by the discussion with @buxxi [here](https://github.com/MichMich/MagicMirror/issues/2471#issuecomment-798600037) this PR adds the possiblity to use another directory than `modules` as parent directory for 3rd-party-modules.

This change simplifies every docker setup because we can now map only the 3rd-party-modules to the host (at the moment the default modules are always included). The new folder should be a subdirectory of the magic mirror main directory.

I used the existing `paths.modules` variable and did minimal code changes.

I see only downsides for modules who are working as module manager for magicmirror:
- MMM-Remote-Control: Minimal backward compatibel code change, I would make a PR in this repo
- mmpm: Will inform the author

Will set this PR to draft and wait for reactions before merging.

### CODE REVIEW

File: js/loader.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ var Loader = (function () {
 			var moduleFolder = config.paths.modules + "/" + module;
 
 			if (defaultModules.indexOf(moduleName) !== -1) {
-				moduleFolder = config.paths.modules + "/default/" + module;
```
Comment (rejas) [inline js/loader.js:94]: is that path already configurable?

Comment (rejas) [reply js/loader.js:94]: if so, why isnt it in the config sample mentioned?

Comment (rejas) [reply js/loader.js:94]: (also: maybe we should add that there?)

Comment (khassel) [reply js/loader.js:94]: path is already configurable, see defaults.js and I tested this. Found nothing documented about this, can add it to the config sample but I dont't know if this a good idea when users who don't know what they are doing are playing with this variable ...

Comment (rejas) [reply js/loader.js:94]: I see. Maybe @MichMich should weigh in on this: if the ALL options should be documented in the config sample?

**Comment by sdetweil** [comment]
be careful moving modules to another path. quite a few will break as they have hard coded modules  in the path to their files.

**Comment by rejas** [comment]
I assume he will keep the current path so that shouldnt be a problem right?

**Comment by khassel** [comment]
I will stop this PR.

Looked at several modules and I see no chance to get it working due to hardcoded things in the modules.

This PR is not the problem, it is backward compatibel if you don't change the modules folder in the config.

But if I would use another folder e.g. in the docker setup, several modules won't work out of the box. So I will stay with the current solution (copying default modules).

If there will be ever a release 3.0 with breaking changes we should discuss to separate default and foreign modules by separate parent folders.

**Comment by khassel** [comment]
I reopened this due to the hint of @MichMich [here](https://github.com/MichMich/MagicMirror/issues/2471#issuecomment-802992169).

> be careful moving modules to another path. quite a few will break as they have hard coded modules in the path to their files.

@sdetweil can you may give some examples?
I had problems with MMM-Remote-Control and MMM-Strava but they are working now.

> I assume he will keep the current path so that shouldnt be a problem right?

@rejas you mean url path or folder path? See changes in `js/server.js`.

The changes in this PR should not affect current setups. Its only the basis for being able to move 3rd-party-modules into another folder.

**Comment by MichMich** [comment]
I'm closing this PR due to inactivity. Feel free to reopen whenever you think it's ready! Thanks!

**Pull request closed without merging**